### PR TITLE
Document leading and {screen}:text-{size} interactions

### PR DESCRIPTION
**Pull request by happycollision**
The change in behavior from 1.x to 2.x was mentioned in the docs, but this makes the information more explicit and also will bring this behavior to the attention of folks who are using tailwind for the first time.

Resolves tailwindlabs/tailwindcss#2603, resolves tailwindlabs/tailwindcss#2852, and resolves tailwindlabs/tailwindcss#2907

> Pr note: The images below represent the rendered output of the changes in this PR. Orange highlight indicates the new sections.

> ### `text-{size}` docs page
CleanShot 2020-11-28 at 14 33 50@2x

> ### `leading-{size}` docs page
CleanShot 2020-11-28 at 14 31 41@2x

> ### upgrade to 2.0 docs page
CleanShot 2020-11-28 at 14 32 35@2x

### CODE REVIEW

File: src/components/Search.js [hunk 1]
```diff
@@ -45,6 +45,7 @@ export function Search() {
     onClose,
     onInput,
     searchButtonRef,
+    actionKey,
```
Comment (simonswiss) [inline src/components/Search.js:48]: This probably doesn't belong in this PR?

**Comment by happycollision** [comment]
Pushed the removal of that errant change. Just doc changes now.

**Comment by adamwathan** [comment]
Thanks, I've simplified it a bit and restricted the documentation of this behavior just to the line-height page to make it easier to maintain but your original commits are all here.
